pep-9999.txt

```diff
@@ -0,0 +1,212 @@
+PEP: 9999
+Title: Support for indexing with keyword arguments
+Version: $Revision$
+Last-Modified: $Date$
+Author: Stefano Borini, Jonathan Fine
+Sponsor: 
+Discussions-To: python-ideas@python.org
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 24-Aug-2020
+Python-Version: 3.10
+Post-History: 
+Resolution: 
+
+Abstract
+========
+
+This PEP proposed extending python to allow keyword-like arguments to be
+accepted during indexing operations. Notations in the form ``a[K=3, R=2]``
+would become legal syntax.  A final strategy will be proposed in terms of
+semantics and implementation.
+
+This PEP is a rework and expansion of PEP-0472, where an extension of the
+indexing operation to support keyword arguments was analysed. PEP-0472 was
+Rejected due to apparent lack of interest back in 2019. However, renewed
+interest has prompted a re-analysis and therefore this PEP.
+
+Background
+==========
+
+PEP-0472 was opened in 2014. The PEP focused on various use cases and was extracted
+from a broad discussion on implementation strategies. The PEP was eventually rejected
+in 2019 [#rejection]_ mostly due to lack of interest despite its 5 years of existence.
+
+However, with the introduction of type hints in PEP-0484 [#pep-0484]_ the
+square bracket notation has been used consistently to enrich the typing
+annotations, e.g. to specify a list of integers as Sequence[int]. As a result,
+a renewed interest in a more flexible syntax that would allow for named
```

Comment:
Perhaps add text along these lines:

Also in the few years there as been expanded growth of packages for data analysis such as pandas and xarray, which use names to describe columns in a table (pandas) or axis in an nd-array (xarray). These packages allow users to access specific data by names, but cannot currently use index notation (``[]``) for this functionality.